pkg/foundation/grpcutil/websocket_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcutil
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/matryer/is"
+)
+
+type testHandler struct {
+	is       *is.I
+	response string
+}
+
+func (h *testHandler) ServeHTTP(w http.ResponseWriter, _ *http.Request) {
+	_, err := w.Write([]byte(h.response))
+	h.is.NoErr(err)
+}
+
+func TestWebSocket_NoUpgradeToWebSocket(t *testing.T) {
```

Comment:
Writing tests for this turned out to be more complex than I thought. A simple http.Handler which we use in this test doesn't work because, as soon as ServeHTTP is done, the request context is cancelled, and so the WebSocket connection is closed. 

I think I'll need to mimic the setup we have when starting Conduit (grpc-gateway and buf-generated code).